test/ExtendedXmlSerializer.Tests.ReportedIssues/Issue361Tests.cs

```diff
@@ -11,26 +14,45 @@ namespace ExtendedXmlSerializer.Tests.ReportedIssues
 	public sealed class Issue361Tests
 	{
 		[Fact]
-		void Verify()
+		void VerifyCustomSerialization_CircularReferenceInsideObjectGraph_DoesNotThrow()
 		{
 			var serializer = new ConfigurationContainer().EnableParameterizedContentWithPropertyAssignments()
-			                                             .Type<AdornedImage>()
-			                                             .CustomSerializer(new AdornedImageSerializer())
+														 .Type<AdornedImage>()
+														 .Register().Serializer().Using(new AdornedImageSerializer())
 			                                             .Create()
 			                                             .ForTesting();
 
-			serializer.Assert(new AdornedImage(),
-			                  @"<?xml version=""1.0"" encoding=""utf-8""?><Issue361Tests-AdornedImage xmlns=""clr-namespace:ExtendedXmlSerializer.Tests.ReportedIssues;assembly=ExtendedXmlSerializer.Tests.ReportedIssues"" />");
+			var image = new AdornedImage();
+
+			serializer.Assert(new DataHolder() { Image1 = image },
+							  @"<?xml version=""1.0"" encoding=""utf-8""?><Issue361Tests-DataHolder xmlns=""clr-namespace:ExtendedXmlSerializer.Tests.ReportedIssues;assembly=ExtendedXmlSerializer.Tests.ReportedIssues""><Name>name</Name><Image1 /><Index>13</Index></Issue361Tests-DataHolder>");
+		}
+
+		[Fact]
+		void VerifyCustomSerialization_CircularReferenceOfRoot_DoesThrow()
+		{
+			var serializer = new ConfigurationContainer().EnableParameterizedContentWithPropertyAssignments()
+														 .Type<AdornedImage>()
+														 .Register().Serializer().Using(new AdornedImageSerializer())
+														 .Create()
+														 .ForTesting();
+
+			var image = new AdornedImage();
+
+			Assert.Throws<CircularReferencesDetectedException >(() => serializer.Serialize(new DataHolder() { Image1 = image, Image2 = image }));
```

Comment:
We want this to _not_ throw, correct?

Reply:
Actually, I believe this should throw, because even if custom serializer is used, it will be called twice. We do not want to throw only if the circular reference is somewhere inside the object sub-graph that will be handled by that custom serializer

Reply:
You are correct.  I am with you now.  I was (obviously) not thorough with the test and thought it was [the class you outlined earlier](https://github.com/ExtendedXmlSerializer/home/issues/361#issuecomment-579668278) with only one image property.